doc/source/myrocks/gap_locks_detection.rst

```diff
@@ -0,0 +1,30 @@
+.. _gap_locks_detection:
+
+===================
+Gap locks detection
+===================
+
+|Percona Server| has implemented `Gap locks
+<https://dev.mysql.com/doc/refman/5.7/en/innodb-locking.html#innodb-gap-locks>`_
+detection in :rn:`5.7.18-14` based on a Facebook |MySQL| patch.
+
+An error will be displayed in the server's error log and returned to the client
+if a transactional storage engine does not support gap locks and a gap lock
+is being attempted.
+
+.. note::
+
+  The gap lock error will not be raised in the condition of a gap lock if the
+  `transation isolation level
+  <https://dev.mysql.com/doc/refman/5.7/en/glossary.html#glos_isolation_level>`_
+  is set to ``READ COMMITTED``.
+
+Running a query which uses gap locks on a storage engine that doesn't have
+implemented support, like *MyRocks* will result in a following error:
+
```

Comment:
I think this can be written a little clearer, maybe something like :
"If a transactional storage engine does not support gap locks (for example MyRocks) and a gap lock is being attempted while the transaction isolation level is either REPEATABLE READ or SERIALIZABLE, the following SQL error will be returned to the client and no actual gap lock will be taken on the effected rows."